src/test/java/games/strategy/engine/framework/startup/login/ClientLoginValidatorTest.java

```diff
@@ -15,26 +16,46 @@
 import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
 
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableMap;
 
 import games.strategy.engine.framework.startup.login.ClientLoginValidator.ErrorMessages;
-import games.strategy.net.IServerMessenger;
+
 
 @RunWith(Enclosed.class)
-public final class ClientLoginValidatorTests {
+public final class ClientLoginValidatorTest {
```

Comment:
Sorry for not being clear about this in my previous comment.  The name of the top-level class should end with `Tests`.  The class that should have been renamed is `MacValidation`.  Therefore, please:

* Revert the name of the top-level class back to `ClientLoginValidatorTests`.
* Rename the `MacValidation` class to `MacValidationTest`.

Reply:
should be done now: 8b4c0c9 + 9db2b75

For Junit4, AFAIK the 'test' suffix is just convention and not actually required (unless a person is using globbing to find test classes). Requiring inner classes to have the specific suffix strikes me as yet one more rule.

Multiple test files merely needs a naming convention and I think we would be done. That does seem easier than the Enclosed mechanism that involves code and a couple of conventions to know.  I'm sorry for the churn, I know I was the one that motivated not having multiple test files, though it does seem like that is probably the best way to go IMO in the end.

Reply:
> For Junit4, AFAIK the 'test' suffix is just convention and not actually required (unless a person is using globbing to find test classes)

Yeah, not a JUnit requirement.  I'd have to dig around for it, but I believe there either is or was something in the Gradle build that relied on this convention.

**EDIT:** Negative on Gradle; it's actually how we suppress the Checkstyle JavadocType rule on test classes (but _not_ test utilities):

```xml
<suppressions>
    <suppress checks="JavadocType" files=".*Tests?\.java"/>
</suppressions>
```

So, yes, we are globbing. :smile:

But it does appear that nested classes without the `Test` suffix wouldn't trigger these types of violations as long as the top-level class had either a `Test` or `Tests` suffix because then the rule is disabled for the entire file.

The `Tests` suffix on nested classes is still required because that's how we avoid duplicate reporting of test results in the test reports.  From `build.gradle`:

```groovy
test {
    exclude '**/*Tests.class'
}
```